bsi_zoo/tests/test_cross_val.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+from scipy import linalg
+
+import pytest
+
+from bsi_zoo.data_generator import get_data
+from bsi_zoo.estimators import (
+    iterative_L1,
+    iterative_L2,
+    iterative_sqrt,
+    iterative_L1_typeII,
+    iterative_L2_typeII,
+    gamma_map
+)
+from bsi_zoo.cross_val import (
+    SpatialCVSolver,
+    TemporalCVSolver
+)
+
+
+@pytest.mark.parametrize("cv_type", ["temporal", "spatial"])
+@pytest.mark.parametrize("orientation_type", ["fixed", "free"])
+@pytest.mark.parametrize("nnz", [3])
+@pytest.mark.parametrize(
+    "estimator,rtol,atol,cov_type,extra_params",
+    [
+        (iterative_L1, 1e-1, 5e-1, "diag", {}),
+        (iterative_L2, 1e-1, 5e-1, "diag", {}),
+        (iterative_sqrt, 1e-1, 5e-1, "diag", {}),
+        (iterative_L1_typeII, 1e-1, 5e-1, "full", {}),
+        (iterative_L2_typeII, 1e-1, 5e-1, "full", {}),
+        (gamma_map, 1e-1, 5e-1, "full", {"update_mode": 1}),
+        (gamma_map, 1e-1, 5e-1, "full", {"update_mode": 2}),
+        (gamma_map, 1e-1, 5e-1, "full", {"update_mode": 3}),
+    ],
+)
+def test_cv(
+    cv_type,
+    estimator,
+    rtol,
+    atol,
+    cov_type,
+    nnz,
+    orientation_type,
+    extra_params,
+):
+    assert cv_type in ["temporal", "spatial"]
+
+    if estimator in [iterative_L1_typeII, iterative_L2_typeII] and cov_type == "full":
+        pytest.skip("iterative L1 type 2 breaks with full covariance")
```

Comment:
@AliHashemi-ai when the cov is full you cannot reduce the number of channels without changing the cov attribute/parameter. How did you deal with this in the past?